streams/jvm/src/main/scala/zio/stream/experimental/platform.scala

```diff
@@ -6,8 +6,54 @@ import zio.stacktracer.TracingImplicits.disableAutoTrace
 import java.io._
 import java.net.InetSocketAddress
 import java.nio.channels.{AsynchronousServerSocketChannel, AsynchronousSocketChannel, CompletionHandler, FileChannel}
-import java.nio.file.Path
+import java.nio.file.{OpenOption, Path}
+import java.nio.file.StandardOpenOption._
 import java.nio.{Buffer, ByteBuffer}
+import java.{util => ju}
+
+trait ZSinkPlatformSpecificConstructors {
+  self: ZSink.type =>
+
+  /**
+   * Uses the provided `Path` to create a [[ZSink]] that consumes byte chunks
+   * and writes them to the `File`. The sink will yield count of bytes written.
+   */
+  final def fromFile(
+    path: => Path,
+    position: Long = 0L,
+    options: Set[OpenOption] = Set(WRITE, TRUNCATE_EXISTING, CREATE)
+  )(implicit
+    trace: ZTraceElement
+  ): ZSink[Any, Throwable, Byte, Throwable, Byte, Long] = {
+    val managedChannel = ZManaged.acquireReleaseWith(
+      ZIO
+        .attemptBlockingInterrupt(
+          FileChannel
+            .open(
+              path,
+              options.foldLeft(
+                new ju.HashSet[OpenOption]()
+              ) { (acc, op) =>
+                acc.add(op)
+                acc
+              } // for avoiding usage of different Java collection converters for different scala versions
+            )
+            .position(position)
+        )
+    )(chan => ZIO.attemptBlocking(chan.close()).orDie)
+
+    val writer = ZSink.unwrapManaged {
+      managedChannel.map { chan =>
+        ZSink.foreachChunk[Any, Throwable, Byte] { byteChunk =>
+          ZIO.attemptBlockingInterrupt {
+            chan.write(ByteBuffer.wrap(byteChunk.toArray))
+          }
+        }
+      }
+    }
+    writer &> ZSink.count
```

Comment:
Do we really need to run two channels in parallel here? All the second channel is doing is keeping track of the count of bytes written. Can't we keep track of that in the `writer` channel and write it as our done value when we are done? That will be much more performant because it will only use one channel instead of two and avoid the overhead of managing the concurrency between them.

Reply:
Done.